Source/debug.cpp

```diff
@@ -565,7 +565,7 @@ std::string DebugCmdLevelUp(const std::string_view parameter)
 	Player &myPlayer = *MyPlayer;
 	std::string cmdLabel = "[givelvl] ";
 
-	const int levels = ParseInt<int>(parameter, /*min=*/1).value_or(1);
+	const int levels = ParseInt<int>(parameter, /*min=*/1, /*max=*/GetMaximumCharacterLevel()).value_or(1);
 	for (int i = 0; i < levels; i++)
 		NetSendCmd(true, CMD_CHEAT_EXPERIENCE);
 	return StrCat(cmdLabel, "New character level: ", myPlayer.getCharacterLevel());

```

Comment:
Okay, sorry for making a mess of your PR. Hopefully this can be the last suggestion. Let's use `std::min()` to clamp the value, incorporate @ephphatha's suggestion, and fix the computation for the new character level.
```suggestion
	const int levels = std::min<int>(ParseInt<int>(parameter, /*min=*/1).value_or(1), GetMaximumCharacterLevel() - myPlayer.getCharacterLevel());
	for (int i = 0; i < levels; i++)
		NetSendCmd(true, CMD_CHEAT_EXPERIENCE);
	return StrCat(cmdLabel, "New character level: ", myPlayer.getCharacterLevel() + levels);
```

Reply:
Haha, no problem :D My fault I didn't check the function definition before. But won't this line:

```const int levels = std::min<int>(ParseInt<int>(parameter, /*min=*/1).value_or(1), GetMaximumCharacterLevel() - myPlayer.getCharacterLevel());```

Make the value always be evaluated to 1 if user enters anything above the current max? Do we want that, or do we want to give him just the max value? Same thing as I've said in @ephphatha comment.

Reply:
I.e: If max lvl is 50, user has lvl 4, won't max in ParseInt be evaluated to 46? Then if user enters 50, this will be value > max, so it will return `ParseIntError:OutOfRange` which will result in the value being converted to 1 instead of clipping it to 50, am I right, or wrong?

Reply:
using std::min in this way lets parseint return a larger value that then gets clamped to the difference between the current player level and the max character level, avoiding the issue where you can't distinguish between parseint returning out of range because the value exceeds the max or is less than the min

Reply:
@ephphatha Oh, nvm, I am blind, I didn't see that this time it is a second parameter to std::min instead of ParseInt. Done!